### ISSUE TITLE
UX/design debt 3

### ISSUE THREAD

**Issue by jorisleker** [kiesraad/abacus#263]
- [x] Consistent page loading experience. The nav menu and footer should not be visible while content is still loading. → #267
  - On `/overview`, you briefly see the page footer right below the top navigation bar whilst the content of the page is loading. 
  - On `/1/input` the entire page is white whilst page is loading: this is how we want it to be. 
- [x] on `/input` allow polling-station-numbers starting with `0` (i.e. `033`). Strip leading zeros before doing lookup of the polling station number in the database (`002` and `02` should evaluate to `2`) → #284
- [x] Hints on how to submit forms with keyboard have been [redesigned](https://www.figma.com/design/zZlFr8tYiRyp4I26sh6eqp/Kiesraad---Abacus-optelsoftware?node-id=812-20649&m=dev) to better show they are keyboard shortcuts. Implement new design. → #284
- [x] in input-grids, when the contents of a cell wraps over two lines, the row-height increases because the text does not fit into remaining cell height. Fix: → #284
  - `input-grid td`'s should have `padding:.5rem 1.5 rem` (iso `1rem 1.5rem`) and `line-height:1.25rem` (iso inherited value of `1.5rem`
  - see screenshots. 1 is before fix, 2 is how it should look
![Scherm­afbeelding 2024-08-22 om 12 31 39](https://github.com/user-attachments/assets/fb09082b-2196-48bf-935f-830a103f31c6)
![Scherm­afbeelding 2024-08-22 om 12 31 07](https://github.com/user-attachments/assets/3f792a37-2727-4c39-8d07-6e7ae3f68993)
- [x] #346
  - See https://github.com/floriankarsten/space-grotesk/issues/44, what do we do with this?
  - → #337 

After #133 is fixed:
- [x] ~Pressing Enter in the last input row returns focus to the first row, it should move focus to the submit button.~ will be fixed in #207 
- [x] Shift+Enter doesn't work (Safari on macOS). → #295
- [x] ~With the tab-key I can highlight the "volgende" button and then submit by hitting the space bar. I can also highlight the navigation items on the left, but there hitting the space bar does not get me to the item I highlighted.~ No fix needed. Spacebar does not submit actions by default. Only `ENTER` does.
- [x] Add custom checkbox component to be able to [better style checkboxes and radiobuttons](https://www.figma.com/design/zZlFr8tYiRyp4I26sh6eqp/Kiesraad---Abacus-optelsoftware?node-id=2498-19792&m=dev) (ie the "confirm to accept warnings" and "recounted" questions) → #292

**Comment by praseodym** [comment]
@jorisleker what do we do about the Space Grotesk font rendering straight quotes as curly quotes? There is an issue opened with the font (https://github.com/floriankarsten/space-grotesk/issues/44) but it doesn't look like there's any progress made on this.

**Comment by praseodym** [comment]
What about renaming our example polling stations so that we don't have quotes there? I can align it with the names from the Figma design.

**Comment by praseodym** [comment]
> The weird thing is, Space Grotesk actually has straight (single and double) quotes. see [Google Fonts glyphs page](https://fonts.google.com/specimen/Space+Grotesk/glyphs). But it looks like the chars are flipped in the fontset. When I copy/paste these straight quotes from the screenshot below, I get these chars: `′` and `″`... So a fix would be to do a replace in all text we render to the front-end, but feels way too hacky.

Those are the Unicode [prime (U+2032)](https://en.wikipedia.org/wiki/Prime_(symbol)) and [double prime (U+2033)](https://en.wikipedia.org/wiki/Prime_(symbol)#Double_prime) symbols. I'm not sure why the font would have those but not proper quote glyphs.

**Comment by jorisleker** [comment]
> What about renaming our example polling stations so that we don't have quotes there? I can align it with the names from the Figma design.
Hiding the issue does not make it go away 😂 But fine with me. 
As long as we have an extended test set with names witn quotes

> Those are the Unicode [prime (U+2032)](https://en.wikipedia.org/wiki/Prime_(symbol)) and [double prime (U+2033)](https://en.wikipedia.org/wiki/Prime_(symbol)#Double_prime) symbols. I'm not sure why the font would have those but not proper quote glyphs.
Ah check. Alternative would be to use [Space Mono](https://fonts.google.com/specimen/Space+Mono/glyphs?preview.layout=grid&query=space+mono) to render straight quotes, because that font has proper quotes. But that feels like an even bigger hack (and requires including an additional font).

**Comment by lkleuver** [comment]
how do we handle keyboard shortcuts?
We can add a listener to buttons for instance,
or I have something laying around that does
```
useKeyboard({
 "shift+enter": doSomething
 });

**Comment by praseodym** [comment]
> how do we handle keyboard shortcuts? We can add a listener to buttons for instance, or I have something laying around that does
> 
> ```
> useKeyboard({
>  "shift+enter": doSomething
>  });
> ```

In the polling station selector we use this, I think that works fine:
https://github.com/kiesraad/abacus/blob/53afc53d79197b9a0d72e7d10fc5a812cc8a4bf9/frontend/app/component/form/polling_station_choice/PollingStationSelector.tsx#L46-L50

**Comment by praseodym** [comment]
> this only triggers when the element has focus, is that the intention of the shortcuts below forms?

If you have an elegant way of making the shortcut work anywhere on the page that would be best. If that's not easily achievable, having it work when focus is on the text boxes is good enough for now.

**Comment by lkleuver** [comment]
This is my current implementation:
https://github.com/kiesraad/abacus/tree/design-debt-3-133/frontend/lib/ui/Button

**Comment by lkleuver** [comment]
but might want to decouple it from a button and create a <Shortcut /> component ?

**Comment by lkleuver** [comment]
I've added code for this:

> With the tab-key I can highlight the "volgende" button and then submit by hitting the space bar. I can also highlight the navigation items on the left, but there hitting the space bar does not get me to the item I highlighted.

But spacebar doesnt normally trigger links from <a> tags. It does now in progress list but wouldn't that make people expect it to work on other links as wel?

**Comment by lkleuver** [comment]
Shift+enter submit works, but the focusing of the next row fires first.
I think we should move the listener to a <Form .../> component

**Comment by jorisleker** [comment]
> But spacebar doesnt normally trigger links from tags. It does now in progress list but wouldn't that make people expect it to work on other links as wel?

spacebar shouldn't work on links. Normal browser behavior is activate focussed action with 'enter', spacebar behaves like pagedown

**Comment by lkleuver** [comment]
@jorisleker if using "ENTER" to go through the inputs, you enter numbers press enter and reach the submit button, it is focused after pressing ENTER on the last input.
What should happen if you press ENTER with the submit button focused ?

**Comment by jorisleker** [comment]
> What should happen if you press ENTER with the submit button focused ?

**I think it makes sense to submit the form when you press `ENTER` with the submit button focussed.**
The thing we wanted to prevent was submitting the form 'to early' / unexpectedly as what happens in OSV. But we have prevented that by splitting up the form in sections, and by adding the 'controleren en opslaan' page/step. 

(am thinking of changing the `SHIFT`+`ENTER` keyboard shortcut to move to the totals fields instead of submitting the form section, because that's an input you always need to complete before submitting, but will put this in a separate ticket)

**Comment by jorisleker** [comment]
- Created separate issue for the space grotesk font rendering issue #346
- Ticket of moving to submit button with `ENTER` from this list because it is addressed in #207 

so this issue can be closed
